File: crates/nexus-agent/src/claude_utils/path.rs
use std::path::PathBuf;

/// Expand `~` to the user's home directory.
pub fn expand_home(path: &str) -> PathBuf {
    if path.starts_with("~/") {
        return nexus_core::paths::home_dir().join(&path[2..]);
    }
    PathBuf::from(path)
}

File: packages/core/src/path.ts
import { homedir } from "node:os";
import { join } from "node:path";

/**
 * Expand a leading `~` or `~/` to the current user's home directory.
 *
 * - `"~"`       -> homedir()
 * - `"~/dev"`   -> join(homedir(), "dev")
 * - `"/abs"`    -> unchanged
 * - `""`        -> unchanged
 * - `"~user"`   -> unchanged (unsupported form)
 */
export function expandTilde(p: string): string {
  if (!p) return p;
  if (p === "~") return homedir();
  if (p.startsWith("~/")) return join(homedir(), p.slice(2));
  return p;
}
